src/autoscaler/api/brokerserver/broker_handler.go

```diff
@@ -104,6 +128,166 @@ func (h *BrokerHandler) CreateServiceInstance(w http.ResponseWriter, r *http.Req
+		updatedDefaultPolicy = ""
+	}
+
+	updatedDefaultPolicyGuid := ""
+	if updatedDefaultPolicy != "" {
+		errResults, valid := h.policyValidator.ValidatePolicy(updatedDefaultPolicy)
+		if !valid {
+			h.logger.Error("failed to validate policy", err, lager.Data{"instanceId": instanceId, "policy": updatedDefaultPolicy})
+			handlers.WriteJSONResponse(w, http.StatusBadRequest, errResults)
+			return
+		}
+		policyGuid, err := uuid.NewV4()
+		if err != nil {
+			h.logger.Error("failed to create policy guid", err, lager.Data{"instanceId": instanceId})
+			writeErrorResponse(w, http.StatusInternalServerError, "Error generating policy guid")
+			return
+		}
+		updatedDefaultPolicyGuid = policyGuid.String()
+	}
+
+	serviceInstance, err := h.bindingdb.GetServiceInstance(instanceId)
+	if err != nil {
+		if err == db.ErrDoesNotExist {
+			h.logger.Error("failed to find service instance to update", err, lager.Data{"instanceId": instanceId})
+			writeErrorResponse(w, http.StatusNotFound, "Failed to find service instance to update")
+			return
+		} else {
+			h.logger.Error("failed to retrieve service instance", err, lager.Data{"instanceId": instanceId})
+			writeErrorResponse(w, http.StatusInternalServerError, "Failed to retrieve service instance")
+			return
+		}
+	}
+
+	updatedServiceInstance := models.ServiceInstance{
+		ServiceInstanceId: serviceInstance.ServiceInstanceId,
+		OrgId:             serviceInstance.OrgId,
+		SpaceId:           serviceInstance.SpaceId,
+		DefaultPolicy:     updatedDefaultPolicy,
+		DefaultPolicyGuid: updatedDefaultPolicyGuid,
+	}
+
+	err = h.bindingdb.UpdateServiceInstance(updatedServiceInstance)
+	if err != nil {
+		h.logger.Error("failed to update service instance", err, lager.Data{"instanceId": instanceId})
+		writeErrorResponse(w, http.StatusInternalServerError, "Error updating service instance")
+		return
+	}
+
+	if serviceInstance.DefaultPolicyGuid != "" {
+		// default policy already exists
+
+		if updatedDefaultPolicyGuid != "" {
+			// new default policy set: replace previous default policies
+			updatedAppIds, err := h.policydb.ReplaceAppPolicies(serviceInstance.DefaultPolicyGuid, updatedDefaultPolicy, updatedDefaultPolicyGuid)
+			if err != nil {
+				h.logger.Error("failed to replace default policies", err, lager.Data{"instanceId": instanceId})
+				writeErrorResponse(w, http.StatusInternalServerError, "Error updating service instance")
+				return
+			}
+
+			// also set the new default policy on bound apps without a policy
+			allBoundApps, err := h.bindingdb.GetAppIdsByInstanceId(serviceInstance.ServiceInstanceId)
+			if err != nil {
+				h.logger.Error("failed to retrieve bound apps", err, lager.Data{"instanceId": instanceId})
+				writeErrorResponse(w, http.StatusInternalServerError, "Error updating service instance")
+				return
+			}
+			newAppsWithDefaultPolicy, err := h.policydb.SetDefaultAppPolicy(allBoundApps, updatedDefaultPolicy, updatedDefaultPolicyGuid)
+			if err != nil {
```

Comment:
If there was a "default policy" defined previously, and `bind-service` will enforce the "default policy" to bound apps ,  I think all the bound apps should have a policy already.  This part is not necessary. 

Furthermore,  if anything error happens here, whether we need to roll back previous update on default policy ?  Maybe we need a cron job to sync the service instance default policy record && policy db  as we did for scheduler ..

Reply:
The policy could have been removed from apps in the mean time. This gives the use to "set the default policy" by calling a service update.

I agree we need a rollback, but I don't think we need a sync. I will add the rollback